radicle-node/src/service/tracking/store.rs

```diff
@@ -0,0 +1,397 @@
+use std::path::Path;
+use std::str::FromStr;
+use std::{fmt, io};
+
+use sqlite as sql;
+use thiserror::Error;
+
+use crate::prelude::Id;
+use crate::service::NodeId;
+
+use super::{Policy, Scope};
+
+#[derive(Error, Debug)]
+pub enum Error {
+    /// I/O error.
+    #[error("i/o error: {0}")]
+    Io(#[from] io::Error),
+    /// An Internal error.
+    #[error("internal error: {0}")]
+    Internal(#[from] sql::Error),
+}
+
+pub type Alias = String;
+
+impl sqlite::BindableWithIndex for Scope {
+    fn bind<I: sql::ParameterIndex>(self, stmt: &mut sql::Statement<'_>, i: I) -> sql::Result<()> {
+        let s = match self {
+            Self::Trusted => "trusted",
+            Self::DelegatesOnly => "delegates-only",
+            Self::All => "all",
+        };
+        s.bind(stmt, i)
+    }
+}
+
+impl TryFrom<&sql::Value> for Scope {
+    type Error = sql::Error;
+
+    fn try_from(value: &sql::Value) -> Result<Self, Self::Error> {
+        let message = Some("invalid remote scope".to_owned());
+
+        match value {
+            sql::Value::String(scope) => Scope::from_str(scope).map_err(|_| sql::Error {
+                code: None,
+                message,
+            }),
+            _ => Err(sql::Error {
+                code: None,
+                message,
+            }),
+        }
+    }
+}
+
+impl sqlite::BindableWithIndex for Policy {
+    fn bind<I: sql::ParameterIndex>(self, stmt: &mut sql::Statement<'_>, i: I) -> sql::Result<()> {
+        match self {
+            Self::Track => "track",
+            Self::Block => "block",
```

Comment:
maybe 'to_string()'?

Reply:
Same as above, not needed yet, but when it is :+1: